app/controllers/sign_up/email_confirmations_controller.rb

```diff
@@ -31,7 +32,10 @@ def process_successful_confirmation
     end
 
     def store_sp_metadata_in_session
-      StoreSpMetadataInSession.new(session:, request_id:).call if request_id.present?
+      if request_id.present?
+        StoreSpMetadataInSession.new(session:, request_id:).call
+        bump_auth_count
+      end
```

Comment:
Minor stylistic comment: Could we make the condition an early return if the condition body is the entire rest of the method?

```suggestion
      return if request_id.blank?
      StoreSpMetadataInSession.new(session:, request_id:).call
      bump_auth_count
```